internal/cmd/manifest_add_test.go

```diff
@@ -0,0 +1,429 @@
+			args: []string{
+				"add", "new-feature",
+				"--default-value", "true",
+				"--description", "A new feature flag",
+			},
+			existingManifest: `{
+				"$schema": "https://raw.githubusercontent.com/open-feature/cli/refs/heads/main/schema/v0/flag-manifest.json",
+				"flags": {}
+			}`,
+			validateResult: func(t *testing.T, fs afero.Fs) {
+				content, err := afero.ReadFile(fs, "flags.json")
+				require.NoError(t, err)
+
+				var manifest map[string]interface{}
+				err = json.Unmarshal(content, &manifest)
+				require.NoError(t, err)
+
+				flags := manifest["flags"].(map[string]interface{})
+				assert.Contains(t, flags, "new-feature")
+
+				flag := flags["new-feature"].(map[string]interface{})
+				assert.Equal(t, "boolean", flag["flagType"])
+				assert.Equal(t, true, flag["defaultValue"])
+				assert.Equal(t, "A new feature flag", flag["description"])
+			},
+		},
+		{
+			name: "add string flag",
+			args: []string{
+				"add", "welcome-message",
+				"--type", "string",
+				"--default-value", "Hello World",
+			},
+			existingManifest: `{
+				"$schema": "https://raw.githubusercontent.com/open-feature/cli/refs/heads/main/schema/v0/flag-manifest.json",
+				"flags": {}
+			}`,
+			validateResult: func(t *testing.T, fs afero.Fs) {
+				content, err := afero.ReadFile(fs, "flags.json")
+				require.NoError(t, err)
+
+				var manifest map[string]interface{}
+				err = json.Unmarshal(content, &manifest)
+				require.NoError(t, err)
+
+				flags := manifest["flags"].(map[string]interface{})
+				flag := flags["welcome-message"].(map[string]interface{})
+				assert.Equal(t, "string", flag["flagType"])
+				assert.Equal(t, "Hello World", flag["defaultValue"])
+			},
+		},
+		{
+			name: "add integer flag",
+			args: []string{
+				"add", "max-retries",
+				"--type", "integer",
+				"--default-value", "5",
+			},
+			existingManifest: `{
+				"$schema": "https://raw.githubusercontent.com/open-feature/cli/refs/heads/main/schema/v0/flag-manifest.json",
+				"flags": {}
+			}`,
+			validateResult: func(t *testing.T, fs afero.Fs) {
+				content, err := afero.ReadFile(fs, "flags.json")
+				require.NoError(t, err)
+
+				var manifest map[string]interface{}
+				err = json.Unmarshal(content, &manifest)
+				require.NoError(t, err)
+
+				flags := manifest["flags"].(map[string]interface{})
+				flag := flags["max-retries"].(map[string]interface{})
+				assert.Equal(t, "integer", flag["flagType"])
+				// JSON unmarshaling converts numbers to float64
+				assert.Equal(t, float64(5), flag["defaultValue"])
+			},
+		},
+		{
+			name: "add float flag",
+			args: []string{
+				"add", "discount-rate",
+				"--type", "float",
+				"--default-value", "0.15",
+			},
+			existingManifest: `{
+				"$schema": "https://raw.githubusercontent.com/open-feature/cli/refs/heads/main/schema/v0/flag-manifest.json",
+				"flags": {}
+			}`,
+			validateResult: func(t *testing.T, fs afero.Fs) {
+				content, err := afero.ReadFile(fs, "flags.json")
```

Comment:
I'm loving all of these tests! Very robust testing! 

This doesn't have to change, but just some food for thought: 
a lot of these values like flags.json, existingManifest, flags: {}, etc. are all repeated in these tests and constant. They may be able to be pulled up into the t.Run below that executes `validateResult`, and that might reduce the lines of code a bit, but it would trade off the ability to customize those values for different test cases if we ever wanted to.